### DISCUSSION TITLE
Getting multiple error with path that are not in the project.

### DISCUSSION THREAD

**Discussion by pkanGitHub** [sqlalchemy/sqlalchemy#9068] [Usage Questions] [answered]
Hello, I am recently working on a project that reads mysql queries from a sql file and output their results onto an excel file. Then I overcome with sqlalchemy modules, everything works, but it will output these error message and crash afterward. I have test out my code and realized it was `read_sql` that is causing the problem. I would like to know if this is the right place for these questions.

```python
def create_connector_engine(database_name):
    engine_uri = f"mysql+mysqlconnector://root:root@mysql-container:3306/{database_name}"
    engine = create_engine(engine_uri)

    return engine

def execute_formatted_file(database, path):
    mydb = create_connector_engine(database)
    with open ('FormattedQueries.sql') as file:
        queries_file = file.read()
        queries = queries_file.splitlines()
        page = 1
        for q in queries:
            pd_query = pd.read_sql(q, mydb)
            try:
                df = pd.DataFrame(pd_query)
                with pd.ExcelWriter(path,mode="a",engine="openpyxl",if_sheet_exists="overlay") as writer:
                    df.to_excel(writer,sheet_name=f"Sheet{page}", index=False)
                    print(f'Query {page} write successfully')
                page += 1

            except Exception as err:
                print(f"Error Occured: {err}")

def main():
    path = 'Result_Data.xlsx'
    writeFile()
    create_workbook(path)  # to create empty excel file.
    database = get_database()
    execute_formatted_file(database, path)
```

The results do execute and write into the excel fine, but I am getting these error messages then the program crash. 

![Screenshot 2023-01-07 at 9 09 34 PM](https://user-images.githubusercontent.com/48925196/211178765-0a4a2c42-cdc8-4521-8071-b3942f027912.png)

Thanks!!

**Comment by CaselIT** [comment, marked as answer]
Hi,

The error seems quite self explanatory. You are running a query that does not return rows, but you are trying to read rows from it, like a DELETE or similar.

**Reply by pkanGitHub** [reply]
Hello, thanks for replying. However, all there is in the sql files are `SELECT` statements. Which should return rows and it did. Also for future reference. How to avoid this error IF I will be reading Update/Delete?
![Screenshot 2023-01-08 at 11 50 17 AM](https://user-images.githubusercontent.com/48925196/211211177-6371067f-f006-4fa7-8bd4-309656deba5d.png)

**Reply by CaselIT** [reply]
You seem to have empty line in the file, but you are not filtering them out. Try to see if ignoring them fixes your issue.

Well, you can't use pandas read_sql to issue update/delete statements, since that expects that rows are returned

**Reply by pkanGitHub** [reply]
OH GOOD EYES!! Yes that fixed the problem! Thank you so much!!

**Reply by CaselIT** [reply]
> Well, you can't use pandas read_sql to issue update/delete statements, since that expects that rows are returned

Sorry I noticed that I wrote "can use" instead of "can't use" in my reply above. I've edited now
